src/data/corpus_base.cpp

```diff
@@ -298,26 +408,28 @@ void CorpusBase::addWeightsToBatch(Ptr<CorpusBatch> batch,
 
 void CorpusBase::initEOS(bool training = true) {
   // Labels fed into sub-batches that are just class-labels, not sequence labels do not require to
-  // add a EOS symbol. Hence decision to add EOS is now based on input stream positions and correspoding
-  // input type.
+  // add a EOS symbol. Hence decision to add EOS is now based on input stream positions and
+  // correspoding input type.
+
+  size_t numStreams = tsv_ ? tsvNumFields_ : paths_.size(); // determine number of streams
 
-  addEOS_.resize(paths_.size(), true);
+  addEOS_.resize(numStreams, true);
   // @TODO: think if this should be checked and processed here or in a validation step in config?
   auto inputTypes = options_->get<std::vector<std::string>>("input-types", {}); // empty list by default
 
   // make sure there is an input type for each path
```

Comment:
for each path -> for each stream (here and below)

Reply:
Done